src/stdune/path.ml

```diff
@@ -490,29 +776,39 @@ let unlink_no_err t = try unlink t with _ -> ()
 
 let build_dir_exists () = is_directory build_dir
 
-let ensure_build_dir_exists () = Local.mkdir_p build_dir
-
-let extend_basename t ~suffix = t ^ suffix
+let ensure_build_dir_exists () =
+  match kind build_dir with
+  | Local p -> Local.mkdir_p p
+  | External p ->
+    let p = External.to_string p in
+    try
+      Unix.mkdir p 0o777
+    with
+    | Unix.Unix_error (EEXIST, _, _) -> ()
+    | Unix.Unix_error (ENOENT, _, _) ->
+      Exn.fatalf "Cannot create external build directory %s. \
```

Comment:
We have `External.mkdir_p` now, should we use it here and in `Path.mkdir_p`?

Reply:
It's not quite the same b/c mkdir_p would re-create the parents. What should this do when foo doesn't exist for example:

```
$ jbuilder build --build-dir /home/ocaml/foo/bar
```

Reply:
I would want `foo` to be created.